tests/test_tables.py

```diff
@@ -149,20 +151,22 @@ def test_get_current_date_from_bad_coupler_res(self):
             get_current_date_from_coupler_res(coupler_res_filename)
 
     def test_get_current_date_from_config(self):
+        rundir = self.make_run_directory('test_rundir')
         config = get_default_config()
         config['namelist']['coupler_nml']['force_date_from_namelist'] = True
         config['namelist']['coupler_nml']['current_date'] = valid_current_date
-        current_date = get_current_date_from_config(config)
+        current_date = get_current_date_from_config(config, os.path.join(rundir, 'INPUT'))
         self.assertEqual(current_date, valid_current_date)
 
-    def test_write_diag_table(self):
+    def test_update_diag_table_for_config(self):
         rundir = self.make_run_directory('test_rundir')
-        diag_table_in = os.path.join(rundir, 'diag_table_in')
-        diag_table_out = os.path.join(rundir, 'diag_table_out')
-        with open(diag_table_in, 'w') as f:
+        input_directory = os.path.join(rundir, 'INPUT')
+        diag_table_filename = os.path.join(rundir, 'diag_table')
+        with open(diag_table_filename, 'w') as f:
             f.write(diag_table_test_in)
-        write_diag_table(config_for_write_diag_table_test, diag_table_in, diag_table_out)
-        with open(diag_table_out) as f:
+        current_date = get_current_date_from_config(config_for_update_diag_table_test, input_directory)
+        update_diag_table_for_config(config_for_update_diag_table_test, current_date, diag_table_filename)
+        with open(diag_table_filename) as f:
```

Comment:
I assume `write_diag_table` got refactored? Just double-checking.

Reply:
Yes, `write_diag_table` was changed to `update_diag_table_for_config` which has slightly different functionality because it changes the diag_table in place instead of writing a new diag_table.

Reply:
Great.